cpp/src/text/edit_distance.cu

```diff
@@ -137,154 +129,157 @@ struct edit_distance_matrix_levenshtein_algorithm {
                                                    rmm::cuda_stream_view stream,
                                                    rmm::device_async_resource_ref mr)
 {
-  cudf::size_type strings_count = strings.size();
-  if (strings_count == 0) {
-    return cudf::make_empty_column(cudf::data_type{cudf::type_to_id<cudf::size_type>()});
-  }
-  CUDF_EXPECTS(strings_count > 1, "the input strings must include at least 2 strings");
-  CUDF_EXPECTS(static_cast<size_t>(strings_count) * static_cast<size_t>(strings_count) <
+  auto const output_type = cudf::data_type{cudf::type_to_id<cudf::size_type>()};
+  if (input.is_empty()) { return cudf::make_empty_column(output_type); }
+  CUDF_EXPECTS(
+    input.size() > 1, "the input strings must include at least 2 strings", std::invalid_argument);
+  CUDF_EXPECTS(input.size() * static_cast<size_t>(input.size()) <
                  static_cast<std::size_t>(std::numeric_limits<cudf::size_type>().max()),
-               "too many strings to create the output column");
+               "too many strings to create the output column",
+               std::overflow_error);
 
   // create device column of the input strings column
-  auto strings_column = cudf::column_device_view::create(strings.parent(), stream);
-  auto d_strings      = *strings_column;
+  auto d_strings = cudf::column_device_view::create(input.parent(), stream);
 
   // Calculate the size of the compute-buffer.
   // We only need memory for half the size of the output matrix since the edit distance calculation
   // is commutative -- `distance(strings[i],strings[j]) == distance(strings[j],strings[i])`
-  cudf::size_type n_upper = (strings_count * (strings_count - 1)) / 2;
-  rmm::device_uvector<std::ptrdiff_t> offsets(n_upper, stream);
-  auto d_offsets = offsets.data();
-  CUDF_CUDA_TRY(cudaMemsetAsync(d_offsets, 0, n_upper * sizeof(std::ptrdiff_t), stream.value()));
-  thrust::for_each_n(
-    rmm::exec_policy(stream),
-    thrust::make_counting_iterator<cudf::size_type>(0),
-    strings_count * strings_count,
-    [d_strings, d_offsets, strings_count] __device__(cudf::size_type idx) {
-      auto const row = idx / strings_count;
-      auto const col = idx % strings_count;
-      if (row >= col) return;  // compute only the top half
-      cudf::string_view const d_str1 =
-        d_strings.is_null(row) ? cudf::string_view{} : d_strings.element<cudf::string_view>(row);
-      cudf::string_view const d_str2 =
-        d_strings.is_null(col) ? cudf::string_view{} : d_strings.element<cudf::string_view>(col);
-      if (d_str1.empty() || d_str2.empty()) { return; }
-      // the temp size needed is 2 integers per character of the shorter string
-      d_offsets[idx - ((row + 1) * (row + 2)) / 2] =
-        (cuda::std::min(d_str1.length(), d_str2.length()) + 1) * 2;
-    });
+  auto const n_upper     = (input.size() * (input.size() - 1)) / 2;
+  auto const output_size = input.size() * input.size();
```

Comment:
Could both overflow?

Reply:
No. This is checked by https://github.com/rapidsai/cudf/pull/18821/files#diff-b7b7d8cbee4653b2285d1a12350bcb68edb6479ec82cce66d28b573af336577dR236